perceval/backends/core/zulip.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2020 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Venu Vardhan Reddy Tekula <venuvardhanreddytekula8@gmail.com>
+#
+
+
+import logging
+
+from grimoirelab_toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient
+
+logger = logging.getLogger(__name__)
+
+CATEGORY_MESSAGE = "message"
+
+
+class Zulip(Backend):
+    """Zulip Backend."""
+
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    def __init__(self, url, stream, api_token, email, origin, tag=None, archive=None, ssl_verify=True):
+        super().__init__(origin, tag, archive, ssl_verify)
+        origin = url
+
+        super.__init__(origin, tag=tag, archive=archive, ssl_verify=True)
+        self.stream = stream
+        self.api_token = api_token
+        self.email = email
+
+        self.client = None
+
+    def fetch(self, category=CATEGORY_MESSAGE):
+        """Fetch the message from the stream."""
+        kwargs = {}
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch the messages."""
+
+        anchor = 'oldest'
+        raw_message = self.client.get_message(anchor)
+        print(raw_message)
+
```

Comment:
The methods to complete the metadata perceval info are missing. Find below possible implementations for them. I guess that Zulip allows to retrieve all history of a channel from the older message to the newest, if this is correct `has_resuming` should return True, otherwise `False` (as similarly done for [Slack](https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/slack.py#L170), which doesn't allow to fetch all history)

```
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not support items resuming
+        """
+        return True
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier of a Zulip item."""
+
+        return str(item['id'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts and coverts the update time from a Zulip item.
+
+        The timestamp is extracted from 'ts' field and converted
+        to a UNIX timestamp.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = float(item['timestamp'])
+
+        return ts
```

Reply:
I see that I didn't copy the diff for the method `metadata_category(item):
```
@@ -160,6 +160,15 @@ class Zulip(Backend):
         ts = float(item['timestamp'])
         return ts
 
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Zulip item.
+
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
```